### PR TITLE
refac(zk): use `Evaluator`

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR has changed to use `Evaluator` to ensure clarity as the files are located under the evaluator directory and include `evaluator.h`.

### CODE REVIEW

File: tachyon/zk/plonk/expressions/evaluator/identifier.h [hunk 1]
```diff
@@ -20,71 +20,83 @@
...
+ public:
+  // Evaluator methods
+  std::string Evaluate(const Expression<F>* input) override {
+    switch (input->type()) {
+      case ExpressionType::kConstant:
+        ss_ << input->ToConstant()->value().ToString();
+        return ss_.str();
+      case ExpressionType::kNegated:
+        ss_ << "(-";
+        ss << Evaluate(input->ToNegated()->expr());
```
Comment (ashjeong) [inline tachyon/zk/plonk/expressions/evaluator/identifier.h]: Please fix this commit and ensure it works by changing instances of `ss` to `ss_`.

**Review by batzor — APPROVED** [review]
LGTM

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Review by GideokKim — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**